WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -17,19 +17,97 @@ private struct QueryRef<Query> {
 /// Subclasses need to implement queriesChanged() and look at activeQueries and
 /// their internal state to decide if they need to act.
 ///
+/// If your `State` type confrorms to the `Codable` protocol, the `QueryStore`
+/// will transparently persist/load State and purge in-memory State when there are no
+/// longer any active `Queries` associated with the `Store`.
+////
+/// You can also manually force writing out the state to disk by calling `persistState()`.
+///
+
 open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
+
+    /// Closure used to log errors. Default implementation calls `NSLog`, clients may find it useful
+    /// to override it to use a custom logging method.
+    var logError: (String) -> Void = { error in
```

Comment:
I suggested a `static var` so it could be overrode per class and not instance, but if it's going to be per-instance, I think a `func` will be more straightforward.

Reply:
Ah, I should have mentioned why it wasn't a `static var` — apparently Swift doesn't allow those on generic classes, hence the instance-level one.